girder/api/v1/resource.py

```diff
@@ -139,6 +140,116 @@ def _getResourceModel(self, kind, funcName=None):
             raise RestException('Invalid resources format.')
         return model
 
+    def _lookupToken(self, token, parentId):
+        """
+        Find a particular child resource by name or throw an exception.
+        :param token: the name of the child resource to find
+        :param parentId: id of the parent to search
+        :returns: the child resource.
+        """
+
+        filterObject = {'name': token}
+        candidateParent = None
+
+        # figure out the parentId's type
+        candidateModelNames = ('collection', 'user', 'folder', 'item')
+        for modelName in candidateModelNames:
```

Comment:
We could make this a lot more efficient by passing the parent type and parent resource (i.e. the dict rather than just the id). Then we could remove these extra lookups, which could get pretty expensive.